src/microbe_stage/CompoundCloudPlane.cs

```diff
@@ -119,7 +119,7 @@ public void UpdatePosition(Int2 newPosition)
     /// <summary>
     ///   Initializes this cloud. cloud2 onwards can be null
     /// </summary>
-    public void Init(FluidSystem fluidSystem, Compound cloud1, Compound? cloud2,
+    public void Init(FluidSystem fluidSystem, int currentIndex, Compound cloud1, Compound? cloud2,
```

Comment:
Please think of a better name for currentIndex, probably renderPriority would be a much better name? Because pure "index" is used as the compound type index within a cloud very heavily.